src/lib.rs

```diff
@@ -1575,6 +1621,116 @@ where
     }
 }
 
+///
+/// A request to revoke a token.
+///
+/// See https://tools.ietf.org/html/rfc7009#section-2.1
+///
+#[derive(Debug)]
+pub struct RevocationRequest<'a, RT, TE>
+where
+    RT: RevocableToken,
+    TE: ErrorResponse,
+{
+    token: RT,
+
+    auth_type: &'a AuthType,
+    client_id: &'a ClientId,
+    client_secret: Option<&'a ClientSecret>,
+    extra_params: Vec<(Cow<'a, str>, Cow<'a, str>)>,
+    revocation_url: Option<&'a RevocationUrl>,
+
+    _phantom: PhantomData<(RT, TE)>,
+}
+
+impl<'a, RT, TE> RevocationRequest<'a, RT, TE>
+where
+    RT: RevocableToken,
+    TE: ErrorResponse + 'static,
+{
+    ///
+    /// Appends an extra param to the token introspect.
+    ///
+    /// This method allows extensions to be used without direct support from
+    /// this crate. If `name` conflicts with a parameter managed by this crate, the
+    /// behavior is undefined. In particular, do not set parameters defined by
+    /// [RFC 6749](https://tools.ietf.org/html/rfc6749) or
+    /// [RFC 7662](https://tools.ietf.org/html/rfc7662).
+    ///
+    /// # Security Warning
+    ///
+    /// Callers should follow the security recommendations for any OAuth2 extensions used with
+    /// this function, which are beyond the scope of
+    /// [RFC 6749](https://tools.ietf.org/html/rfc6749).
+    ///
+    pub fn add_extra_param<N, V>(mut self, name: N, value: V) -> Self
+    where
+        N: Into<Cow<'a, str>>,
+        V: Into<Cow<'a, str>>,
+    {
+        self.extra_params.push((name.into(), value.into()));
+        self
+    }
+
+    fn prepare_request<RE>(self) -> Result<HttpRequest, RequestTokenError<RE, TE>>
+    where
+        RE: Error + 'static,
+    {
+        let mut params: Vec<(&str, &str)> = vec![("token", self.token.secret())];
+        params.push(("token_type_hint", self.token.token_type_hint()));
+
+        Ok(endpoint_request(
+            self.auth_type,
+            self.client_id,
+            self.client_secret,
+            &self.extra_params,
+            None,
+            None,
+            self.revocation_url
```

Comment:
I noticed that the [RFC](https://tools.ietf.org/html/rfc7009#section-2) says `Clients MUST verify that the URL is an HTTPS URL`. we should probably fail here for other schemes